File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

long long solution(long long n)
{
    long long ans = 0;
    ans = sqrt(n);

    if (ans * ans == n)
        return pow(ans+1, 2);

    return -1;
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        double sqrted = Math.sqrt(n);
        if (sqrted * sqrted == n && sqrted % 1 == 0.0) {
            double pow = Math.pow(sqrted + 1, 2);
            return (long) pow;
        }
        return -1;
    }
}